pkg/arvo/sys/hoon.hoon

```diff
@@ -226,6 +226,33 @@
     *   (add (mod b 2) (mul $(b (div b 2)) 2))
   ==
 ::
+++  pin
+  ~/  %pin
+  ::    sub axis test
+  ::
+  ::  checks whether {b} addresses a subtree of the tree addressed by {a}.
+  |=  [a=@ b=@]
+  ?<  =(0 a)
+  ?<  =(0 b)
+  |-  ^-  ?
+  ?:  =(a 1)  %&
```

Comment:
I think `+pin` should be total over non-zero atoms. For that to be the case, this should be

```
?:  =(a 1)  &
?:  =(b 1)  |
```